backend/ownerSizeTotalWorker.test.js

```diff
@@ -0,0 +1,288 @@
+        switch (subop) {
+          case 0: {
+            console.log('Simulated new event on', contract.id)
+            await contract.newMessage()
+            break
+          }
+          case 1: {
+            console.log('Simulated KV update on', contract.id)
+            await contract.newKvValue()
+            break
+          }
+          case 3: {
+            const subContract = new Contract(contract)
+            console.log('Simulated new subcontract for', contract.id, 'id', subContract.id, 'ultimate owner', subContract.ultimateOwner.id)
+            await subContract.init()
+            allContracts.push(subContract)
+            break
+          }
+        }
+        break
+      }
+      case 6: {
+        const wait = (0, Math.random)() * 100 | 0
+        console.log('Random wait', wait)
+        await new Promise(resolve => setTimeout(resolve, wait))
+        break
+      }
+      case 7: {
+        console.log('Simulating forcible computation')
+        await new Promise((resolve, reject) => {
+          const mc = new MessageChannel()
+          mc.port2.onmessage = (event) => {
+            const [success: boolean, result: any] = event.data
+            if (success) return resolve()
+            reject(result)
+          }
+          mc.port2.onmessageerror = () => {
+            reject(Error('Message error'))
+          }
+          worker.postMessage([mc.port1, 'backend/server/computeSizeTaskDeltas'], [mc.port1])
+        })
+        break
+      }
+      case 8: {
+        console.log('Simulating server restart')
+        await new Promise(resolve => setTimeout(resolve, 250))
+        await worker.terminate()
+        worker = new Worker(join(__dirname, 'ownerSizeTotalWorker.js'))
+        await new Promise((resolve, reject) => {
+          worker.on('message', (msg) => {
+            if (msg === 'ready') resolve()
+          })
+          worker.on('error', reject)
+        })
+        break
+      }
+    }
+
+    yield
+  }
+
+  // Wait for transactions to be written
+  await new Promise((resolve, reject) => {
+    const mc = new MessageChannel()
+    mc.port2.onmessage = (event) => {
+      const [success: boolean, result: any] = event.data
+      if (success) return resolve()
+      reject(result)
+    }
+    mc.port2.onmessageerror = () => {
+      reject(Error('Message error'))
+    }
+    worker.postMessage([mc.port1, 'backend/server/computeSizeTaskDeltas'], [mc.port1])
+  })
+  return mainContracts
+}
+
+describe('Owner total size computation fuzzing', () => {
+  before(async () => {
+    await initDB()
+    await workerReady
+  })
+
+  after(async () => {
+    await worker.terminate()
+  })
+
+  it('Simulated events have the expected size', async () => {
+    const iterations = (0, Math.random)() * 4096 | 512
+    const iterator = randomOp(iterations)
+    let v
+    for (;;) {
+      v = await iterator.next()
+      if (v.done) break
+    }
+
+    const contracts = v.value
```

Comment:
I'm a little curious as to why a generator like this was used at all to step through the simulation...

What is the value of using `yield` inside of `randomOp` and then looping here? Why not just let the randomOp's while loop spin and then get the final result?

Reply:
Because simulations normally are steps, so a generator seemed appropriate, and also allows for adding additional checks or other actions in the future.